sklearn/tests/test_min_dependencies_readme.py

```diff
@@ -28,7 +28,7 @@ def test_min_dependencies_readme():
         + r"( [0-9]+\.[0-9]+(\.[0-9]+)?)"
     )
 
-    readme_path = Path(sklearn.__path__[0]).parents[0]
+    readme_path = Path(sklearn.__file__).parent.parent
```

Comment:
Can we have a separate PR for all the `sklearn.__file__` changes?

(To reduce the diff in this PR)

Reply:
This is doable, although a bit more work. This would mean:
- a first PR that adds Meson support without any CI
- a second PR that adds a build with one build testing the meson build and the `sklearn.__path__` work-around. I opened https://github.com/mesonbuild/meson-python/issues/557 about it by the way.